geomesa-convert/geomesa-convert-common/src/main/scala/org/locationtech/geomesa/convert/Transformers.scala

```diff
@@ -482,5 +481,35 @@ class MapListFunctionFactory extends TransformerFunctionFactory {
         }.toMap.asJava
     }
   }
+}
+
+class CastFunctionFactory extends TransformerFunctionFactory {
+  override def functions = Seq(string2double, string2int, string2float, string2long, string2boolean)
 
+  val string2double = TransformerFn("string2double") {
+    args => {
+      if (isEmptyOrNull(args(0))) { args(1) } else args(0).asInstanceOf[String].toDouble
+    }
+  }
+  val string2int = TransformerFn("string2int") {
+    args => {
+      if (isEmptyOrNull(args(0))) { args(1) } else args(0).asInstanceOf[String].toInt
+    }
+  }
+  val string2float = TransformerFn("string2float") {
+    args => {
+      if (isEmptyOrNull(args(0))) { args(1) } else args(0).asInstanceOf[String].toFloat
```

Comment:
I guess another issue is what if the string has something, but it's not a valid e.g. float? We should probably wrap the else statement in a try/catch, and return the default if it fails.